File: Andromeda.AvaloniaApp.FSharp/Converter/PathToBitmapConverter.cs
using Avalonia;
using Avalonia.Controls;
using Avalonia.Data.Converters;
using Avalonia.Media.Imaging;
using Microsoft.FSharp.Core;
using System;
using System.Globalization;

namespace Andromeda.AvaloniaApp.Converter
{
    public class PathToBitmapConverter : IValueConverter
    {
        public object Convert(object value, Type targetType,
            object parameter, CultureInfo culture)
        {
            if (value != null) {
                // Unwrap optiontype if necessary
                if (value is FSharpOption<string>) {
                    value = (value as FSharpOption<string>).Value;
                }
                return new Bitmap((string) value);
            } else {
                return null;
            }
        }

        public object ConvertBack(object value, Type targetType,
            object parameter, CultureInfo culture)
        {
            var bitmap = (Bitmap) value;
            return bitmap.ToString();
        }
    }
}

File: Andromeda.AvaloniaApp.FSharp/Converter/PathToBitmapConverter.fs
namespace Andromeda.AvaloniaApp.FSharp.Converter

open Avalonia.Data.Converters
open Avalonia.Media.Imaging
open Microsoft.FSharp.Core

type PathToBitmapConverter () =
    interface IValueConverter with
        member __.Convert(value, _, _, _) =
            match value with
            | null -> null
            | value ->
                // Unwrap optiontype if necessary
                let value =
                    match value with
                    | :? Option<string> as value ->
                        value.Value
                    | :? string as value->
                        value
                    | _ -> invalidArg "value" "no float or float option"
                new Bitmap(value) :> obj

        member __.ConvertBack(value, _, _, _) =
            (value :?> Bitmap).ToString() :> obj
